### PR TITLE
Hosting Metrics: Add requests by HTTP response code grouped bar chart first pass

### PR DESCRIPTION
**Pull request by sejas**
See https://github.com/Automattic/dotcom-forge/issues/3356

## Proposed Changes

* It introduces a new uPlot Grouped bar chart to hosting metrics and storybook

## Testing Instructions

* Navigate to `/site-monitoring/${siteSlug}`
* Scroll down
* Observe the new bar chart

**Storybook**

- Run `yarn storybook:start`
- Access http://localhost:6006/?path=/story/packages-components-chart-uplot--bar-chart-uplot
- Observe the Grouped Bar Chart looks like the design

### Screenshots

Screenshot 2023-08-04 at 17 54 04

Screenshot 2023-08-03 at 19 12 08

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by sejas** [comment]
These files are an adaptation of [uPlot demo](https://github.com/leeoniya/uPlot/blob/606c59dc656f73dea977e8337f357af7de4f0e6e/demos/multi-bars.html#L36-L38):
- `packages/components/src/chart-uplot/lib/quadtree.js`
- `packages/components/src/chart-uplot/uplot-plugins/multi-bars.ts`

File: client/my-sites/site-monitoring/components/site-monitoring-bar-chart/use-metrics-bar-chart-data.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { useMemo } from 'react';
+import { TimeRange } from '../../main';
+import { PeriodData, useSiteMetricsQuery } from '../../use-metrics-query';
+
+const STATUS_CODES = [ 200, 401, 400, 404, 500 ];
```
Comment (danielbachhuber) [inline client/my-sites/site-monitoring/components/site-monitoring-bar-chart/use-metrics-bar-chart-data.ts:5]: I guess we need to figure out which status codes we want to show. Showing all of them probably isn't realistic.

**Review by danielbachhuber — APPROVED** [review]
Looks great, @sejas. Thanks for your work on this!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8684041

Thank you @sejas for including a screenshot in the description! This is really helpful for our translators.

File: client/my-sites/site-monitoring/components/site-monitoring-bar-chart/use-metrics-bar-chart-data.ts (hunk 1 above)
Comment (sejas) [reply client/my-sites/site-monitoring/components/site-monitoring-bar-chart/use-metrics-bar-chart-data.ts:5]: Yes, I had to decide between
Dynamically extract the status codes returned by the API, which in my dummy sites were just three `200`, `404`, and '400` and in the chart, status codes with 0 will not appear.

Versus

Displaying "selected" status codes. I took this path to make the chart look like the design.

**Comment by sejas** [comment]
> Looks great, @sejas. Thanks for your work on this!

Thanks for reviewing and deploying 🙌

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8684041).

**Pull request merged**